src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -648,6 +650,89 @@ class ActorIdStringGenerator : public Generator<std::string> {
     std::string _actorId;
 };
 
+/** `{^Now: {}}` */
+class NowGenerator : public Generator<bsoncxx::types::b_date> {
+public:
+    NowGenerator(const Node& node, GeneratorArgs generatorArgs) {}
+    bsoncxx::types::b_date evaluate() override {
+        return bsoncxx::types::b_date{std::chrono::system_clock::now()};
+    }
+};
+
+// The 2 formats also cover "%Y-%m-%d", timezones are not parsed,
+// %z, %Q are output only https://www.boost.org/doc/libs/1_75_0/doc/html/date_time/date_time_io.html.
+const std::locale formats[] = {
```

Comment:
this can be a `std::array` since you know the size at compile-time, and you can use that instead of `formats_n` as well. Moreover C++17 lets you use auto-deduction:

```c++
const std::array formats = { .... }
// or std::make_array would work
```